### PR TITLE
[FL-3757] NFC: fix application opening from browser

### PR DESCRIPTION
**Pull request by RebornedBrain**
# What's new

- Previously when we tried to open .nfc file, where we don't support emulation, from browser then app crashes. Now it will show saved menu

# Verification 

- Go to browser and open any ,nfc file of type which doesn't support emulation. Result: saved menu will be opened.
- Go to browser and open any ,nfc file of type which supports emulation. Result: emulation will be started.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by gornekich — DISMISSED** [review]

**Comment by xMasterX** [comment]
Hi!
These changes does not fix the issue fully, 
Delete button causes crash
Rename button causes exit into main NFC app - incorrect logic

All required fixes are done in attached patch file, apply before merging
Thanks

[287239.patch](https://github.com/flipperdevices/flipperzero-firmware/files/14104154/287239.patch)

**Comment by xMasterX** [comment]
And this patch after first one too
[fe05c6.patch](https://github.com/flipperdevices/flipperzero-firmware/files/14111739/fe05c6.patch)

File: applications/main/nfc/scenes/nfc_scene_save_success.c [hunk 1]
```diff
@@ -31,9 +31,23 @@ bool nfc_scene_save_success_on_event(void* context, SceneManagerEvent event) {
...
+                consumed = scene_manager_search_and_switch_to_previous_scene(
+                    nfc->scene_manager, NfcSceneFileSelect);
+                if(!consumed) {
+                    consumed = scene_manager_search_and_switch_to_previous_scene(
+                        nfc->scene_manager, NfcSceneSavedMenu);
+                    if(!consumed) {
+                        consumed = scene_manager_search_and_switch_to_another_scene(
+                            nfc->scene_manager, NfcSceneFileSelect);
+                    }
+                }
```
Comment (gornekich) [inline applications/main/nfc/scenes/nfc_scene_save_success.c]: Not in the same style as above.

**Review by gornekich — CHANGES_REQUESTED** [review]
It's more clear if all navigation logic is written in one style. Apply patch
[scene_save_success_cleanup.txt](https://github.com/flipperdevices/flipperzero-firmware/files/14171419/scene_save_success_cleanup.txt)

File: applications/main/nfc/scenes/nfc_scene_save_success.c (hunk 1 above)
Comment (RebornedBrain) [reply applications/main/nfc/scenes/nfc_scene_save_success.c]: Done

**Review by gornekich — APPROVED** [review]

**Review by hedger — APPROVED** [review]

**Pull request merged**